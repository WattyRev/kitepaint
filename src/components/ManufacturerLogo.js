import styled from "styled-components";

/**
 * Styling for the manufacturer logo
 */
const ManufacturerLogo = styled.div`
  background-image: url(${props => props.src});
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center;
  display: block;
  width: ${props => props.size || 70}px;
  height: ${props => props.size || 70}px;
  margin: ${props => (props.noMargin ? 0 : "8px 8px 0 0")};
  border: 1px solid ${props => props.theme.colors.gray};
  border-radius: 4px;
  box-sizing: border-box;
  background-color: ${props => props.theme.colors.white};
`;

export default ManufacturerLogo;
