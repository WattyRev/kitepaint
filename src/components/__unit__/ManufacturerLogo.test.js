import React from "react";
import { mount } from "enzyme";
import Theme from "../../theme";
import ManufacturerLogo from "../ManufacturerLogo";

describe("ManufacturerLogo", () => {
  it("renders", () => {
    mount(<ManufacturerLogo theme={Theme} />);
  });
  it("uses the src as the background-image", () => {
    const wrapper = mount(
      <ManufacturerLogo theme={Theme} src="my-image.jpg" />
    );
    expect(wrapper).toHaveStyleRule("background-image", "url(my-image.jpg)");
  });
  it("sets the width/height to 70px by default", () => {
    const wrapper = mount(<ManufacturerLogo theme={Theme} />);
    expect(wrapper).toHaveStyleRule("width", "70px");
    expect(wrapper).toHaveStyleRule("height", "70px");
  });
  it("sets the width/height according to the provided size property", () => {
    const wrapper = mount(<ManufacturerLogo theme={Theme} size={16} />);
    expect(wrapper).toHaveStyleRule("width", "16px");
    expect(wrapper).toHaveStyleRule("height", "16px");
  });
  it("sets the margin by default", () => {
    const wrapper = mount(<ManufacturerLogo theme={Theme} />);
    expect(wrapper).toHaveStyleRule("margin", "8px 8px 0 0");
  });
  it("clears the margin if noMargin is provided", () => {
    const wrapper = mount(<ManufacturerLogo theme={Theme} noMargin />);
    expect(wrapper).toHaveStyleRule("margin", "0");
  });
});
