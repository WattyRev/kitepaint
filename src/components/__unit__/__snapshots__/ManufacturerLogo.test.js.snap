// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManufacturerLogo renders 1`] = `
.c0 {
  background-image: url();
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center;
  display: block;
  width: 70px;
  height: 70px;
  margin: 8px 8px 0 0;
  border: 1px solid #AAAAAA;
  border-radius: 4px;
  box-sizing: border-box;
  background-color: #FFFFFF;
}

<styled.div
  theme={
    Object {
      "colors": Object {
        "aqua": "#7FDBFF",
        "black": "#111111",
        "blue": "#0074D9",
        "blueDarker": "#005daa",
        "blueLighter": "#f0fbff",
        "fuchsia": "#F012BE",
        "gray": "#AAAAAA",
        "grayDark": "#5e5e5e",
        "grayDarker": "#3a3a3a",
        "green": "#2ECC40",
        "lime": "#01FF70",
        "maroon": "#85144B",
        "navy": "#001F3F",
        "olive": "#3D9970",
        "orange": "#FF851B",
        "purple": "#B10DC9",
        "red": "#FF4136",
        "silver": "#DDDDDD",
        "teal": "#39CCCC",
        "white": "#FFFFFF",
        "yellow": "#FFDC00",
      },
      "patterns": Object {
        "transparencyBackground": Array [
          "
  background-image: linear-gradient(45deg, ",
          "#AAAAAA",
          " 25%, transparent 25%),
    linear-gradient(-45deg, ",
          "#AAAAAA",
          " 25%, transparent 25%),
    linear-gradient(45deg, transparent 75%, ",
          "#AAAAAA",
          " 75%),
    linear-gradient(-45deg, transparent 75%, ",
          "#AAAAAA",
          " 75%);
  background-color: ",
          "#5e5e5e",
          ";
  background-size: 20px 20px;
  background-position: 0 0, 0 10px, 10px -10px, -10px 0px;
",
        ],
      },
    }
  }
>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-kpOJdX",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  background-image: url(",
            [Function],
            ");
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center;
  display: block;
  width: ",
            [Function],
            "px;
  height: ",
            [Function],
            "px;
  margin: ",
            [Function],
            ";
  border: 1px solid ",
            [Function],
            ";
  border-radius: 4px;
  box-sizing: border-box;
  background-color: ",
            [Function],
            ";
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-kpOJdX",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    theme={
      Object {
        "colors": Object {
          "aqua": "#7FDBFF",
          "black": "#111111",
          "blue": "#0074D9",
          "blueDarker": "#005daa",
          "blueLighter": "#f0fbff",
          "fuchsia": "#F012BE",
          "gray": "#AAAAAA",
          "grayDark": "#5e5e5e",
          "grayDarker": "#3a3a3a",
          "green": "#2ECC40",
          "lime": "#01FF70",
          "maroon": "#85144B",
          "navy": "#001F3F",
          "olive": "#3D9970",
          "orange": "#FF851B",
          "purple": "#B10DC9",
          "red": "#FF4136",
          "silver": "#DDDDDD",
          "teal": "#39CCCC",
          "white": "#FFFFFF",
          "yellow": "#FFDC00",
        },
        "patterns": Object {
          "transparencyBackground": Array [
            "
  background-image: linear-gradient(45deg, ",
            "#AAAAAA",
            " 25%, transparent 25%),
    linear-gradient(-45deg, ",
            "#AAAAAA",
            " 25%, transparent 25%),
    linear-gradient(45deg, transparent 75%, ",
            "#AAAAAA",
            " 75%),
    linear-gradient(-45deg, transparent 75%, ",
            "#AAAAAA",
            " 75%);
  background-color: ",
            "#5e5e5e",
            ";
  background-size: 20px 20px;
  background-position: 0 0, 0 10px, 10px -10px, -10px 0px;
",
          ],
        },
      }
    }
  >
    <div
      className="c0"
    />
  </StyledComponent>
</styled.div>
`;
